### PR TITLE
fix: Incorrect aggregation of empty groups after slice

### PR DESCRIPTION
**Pull request by nameexhaustion**
Returned previously filtered values instead of NULL.

Fixes https://github.com/pola-rs/polars/issues/20115

### CODE REVIEW

File: crates/polars-expr/src/expressions/slice.rs [hunk 1]
```diff
@@ -110,6 +110,11 @@ impl PhysicalExpr for SliceExpr {
                 .collect::<PolarsResult<Vec<_>>>()
         })?;
         let mut ac = results.pop().unwrap();
+
+        if ac.is_aggregated() {
+            polars_bail!(InvalidOperation: "cannot slice() an aggregated value")
```
Comment (nameexhaustion) [inline crates/polars-expr/src/expressions/slice.rs:115]: drive-by - I noticed we currently silently ignore `slice()` after aggregations - I've made this raise instead.

File: crates/polars-core/src/frame/column/mod.rs [hunk 2]
```diff
@@ -657,7 +657,8 @@ impl Column {
                 let mut s = scalar_col.take_materialized_series().rechunk();
                 // SAFETY: We perform a compute_len afterwards.
                 let chunks = unsafe { s.chunks_mut() };
-                chunks[0].with_validity(Some(validity));
```
Comment (nameexhaustion) [inline crates/polars-core/src/frame/column/mod.rs:660]: this function isn't in-place and we didn't assign back 😄

Comment (coastalwhite) [reply crates/polars-core/src/frame/column/mod.rs:660]: Auch...
